stl/inc/compare

```diff
@@ -340,9 +341,503 @@ struct compare_three_way {
+                return {_St::_Adl,
+                    noexcept(static_cast<strong_ordering>(strong_order(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
+            } else if constexpr (floating_point<decay_t<_Ty1>>) {
+                return {_St::_Floating, true};
+            } else if constexpr (_Can_compare_three_way<_Ty1, _Ty2>) {
+                return {_St::_Three, noexcept(static_cast<strong_ordering>(
+                                         compare_three_way{}(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
+            } else {
+                return {_St::_None};
+            }
+        }
+
+        template <class _Ty1, class _Ty2>
+        static constexpr _Choice_t<_St> _Choice = _Choose<_Ty1, _Ty2>();
+
+    public:
+        // clang-format off
+        template <class _Ty1, class _Ty2>
+            requires (_Choice<_Ty1, _Ty2>._Strategy != _St::_None)
+        _NODISCARD constexpr strong_ordering operator()(_Ty1&& _Left, _Ty2&& _Right) const
+            noexcept(_Choice<_Ty1, _Ty2>._No_throw) {
+            // clang-format on
+            constexpr _St _Strat = _Choice<_Ty1, _Ty2>._Strategy;
+            if constexpr (_Strat == _St::_Adl) {
+                return static_cast<strong_ordering>(strong_order(_Left, _Right));
+            } else if constexpr (_Strat == _St::_Floating) {
+                using _Floating_type = decay_t<_Ty1>;
+                using _Traits        = _Floating_type_traits<_Floating_type>;
+                using _Uint_type     = typename _Traits::_Uint_type;
+
+                auto _Left_uint  = _STD bit_cast<_Uint_type>(_Left);
+                auto _Right_uint = _STD bit_cast<_Uint_type>(_Right);
+
+                // 1. Ultra-fast path: equal representations are equal.
+                if (_Left_uint == _Right_uint) {
+                    return strong_ordering::equal;
+                }
+
+                // 2. Examine the sign bits.
+                const bool _Left_negative  = (_Left_uint & _Traits::_Shifted_sign_mask) != 0;
+                const bool _Right_negative = (_Right_uint & _Traits::_Shifted_sign_mask) != 0;
+
+                // 3. Fast path: any negative value is less than any positive value.
+                if (_Left_negative != _Right_negative) {
+                    return _Left_negative ? strong_ordering::less : strong_ordering::greater;
+                }
+
+                // 4. Clear the (identical) sign bits. We've already stored them for use at the end.
+                _Left_uint &= ~_Traits::_Shifted_sign_mask;
+                _Right_uint &= ~_Traits::_Shifted_sign_mask;
+
+                // 5. Perform the final comparison, reversed for negative values.
+                // (For example, 1.5 is less than 2.5, but -1.5 is greater than -2.5.)
+                if (_Left_negative) {
+                    return _Right_uint <=> _Left_uint;
+                } else {
+                    return _Left_uint <=> _Right_uint;
+                }
+            } else {

```

Comment:
I think this should be an `else if`. The `static_assert` below kind of obscures that we should never reach the `else`, which is quite clear via the `static_assert(_Always_false<_Ty>, "should be unreachable");` that is use in the other CPOs

Reply:
Thanks for catching this - I was trying to simplify the original pattern, but obscured it, *and* introduced inconsistency between files while doing so. Changed all occurrences to use `_Always_false` as you recommended.